ITL-4 Implement base state of todo card

EmanuelRusuu opened the pull request.

EmanuelRusuu committed:
ITL-21 Subtask: When the user completes a todo, animate a checkmark and move the completed todo item to the bottom of the todo list

EmanuelRusuu committed:
Updated styling to figma

EmanuelRusuu committed:
ITL-22 Make an input field and design it

- added an input field
- search icon
- search button
- styling according to figma

EmanuelRusuu committed:
ITL-23 Make the input field to work, search the todo list for specific words

- the search bar works;
- it searches and compares the input value of the search bar with a todo`s title and text.

kincsokiss commented on todoapp/src/components/EmptyState.vue:
```diff
@@ -1,13 +1,9 @@
 
 <template>
-  <div v-if="todos.length < 1" class="no-todos">
+  <div class="no-todos">
```
Please check below the `.no-todos-p` in the <styles> section and remove the `text-align` and `color` from the `media` part, because those are set generally as well.

kincsokiss commented on todoapp/src/components/TodoList.vue:
```diff
@@ -29,6 +31,10 @@ onMounted(() => {
   if (storedTodos !== null) {
```
You can remove the `!== null` from the condition, the reason will be the same.

kincsokiss commented on todoapp/src/components/Todo.vue:
```diff
@@ -42,6 +42,20 @@ function setPriority(todo: TodoType, number: number) {
 }
 
+function markTodoStatus(todo: TodoType) {
+  if (todo.status) {
```
I think you should have one function to set the Todo status done or not done, and in the implementation of that function should decide what should happen based on the returned todo status.

EmanuelRusuu committed:
[ITL-4] - Fixed PR issues

- placed the template at the top, script in the middle and style at the bottom
- removed duplicated code in style
- removed "!== null" code in checking localStorage todos, as it was unnecessary
- solved the todo status function

EmanuelRusuu committed:
Merge branch 'feature/ITL-6-add-sorting-functionality' into develop

EmanuelRusuu committed:
[ITL-4] - Fixed rebase conflicts

kincsokiss commented on todoapp/src/components/Todo.vue:
```diff
@@ -244,11 +266,15 @@ function textEdit(todo: TodoType) {
+  display: block;
 }
 
 .todo-created_at {
```
Please use `-` instead of `_` in the name of the class.
